src/main/java/de/tum/in/www1/artemis/repository/settings/UserIdeMappingRepository.java

```diff
@@ -0,0 +1,25 @@
+package de.tum.in.www1.artemis.repository.settings;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_CORE;
+
+import java.util.List;
+
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Repository;
+
+import de.tum.in.www1.artemis.domain.settings.ide.Ide;
+import de.tum.in.www1.artemis.domain.settings.ide.UserIdeMapping;
+import de.tum.in.www1.artemis.domain.settings.ide.UserIdeMappingId;
+import de.tum.in.www1.artemis.repository.base.ArtemisJpaRepository;
+
+/**
+ * Spring Data repository for the UserIdeMapping entity.
+ */
+@Profile(PROFILE_CORE)
+@Repository
+public interface UserIdeMappingRepository extends ArtemisJpaRepository<UserIdeMapping, UserIdeMappingId> {
+
+    List<UserIdeMapping> findAllByUserId(Long userId);
+
+    List<UserIdeMapping> findAllByIde(Ide ide);
```

Comment:
You only use this to check if it is empty. Since this could then lead to the unnecessary loading of several thousand entries in the future, please change this to a count query.

Reply:
👍 changed to exists check now